presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedClusterStatsResource.java

```diff
@@ -79,25 +111,77 @@ private String getResourceFilePath(String fileName)
         assertTrue(clusterStats.getActiveWorkers() > 0);
-        assertEquals(clusterStats.getRunningQueries(), 1);
+        assertEquals(clusterStats.getRunningQueries(), 3);
         assertEquals(clusterStats.getBlockedQueries(), 0);
-        assertEquals(clusterStats.getQueuedQueries(), 0);
+        assertEquals(clusterStats.getQueuedQueries(), 1);
+        assertEquals(clusterStats.getAdjustedQueueSize(), 0);
+    }
+
+    private void waitForGlobalQueryViewInCoordinator(int numberOfRunningQueries, QueryState state, TestingPrestoServer coordinator, long timeoutInMillis)
+            throws InterruptedException, TimeoutException
+    {
+        long deadline = System.currentTimeMillis() + timeoutInMillis;
+        int globalQueryCount = 0;
+        while (System.currentTimeMillis() < deadline) {
+            Map<ResourceGroupId, ResourceGroupRuntimeInfo> resourceGroupRuntimeInfoSnapshot = coordinator.getResourceGroupManager().get().getResourceGroupRuntimeInfosSnapshot();
+            ResourceGroupRuntimeInfo resourceGroupRuntimeInfo = resourceGroupRuntimeInfoSnapshot.get(new ResourceGroupId(RESOURCE_GROUP_GLOBAL));
+            if (resourceGroupRuntimeInfo == null) {
+                continue;
+            }
+            globalQueryCount = getQueryCount(state, resourceGroupRuntimeInfo);
+            if (globalQueryCount == numberOfRunningQueries) {
+                return;
+            }
+            sleep(100);
+        }
+        throw new TimeoutException(format("Global Query Count: %s after %s ms", globalQueryCount, timeoutInMillis));
+    }
+
+    private int getQueryCount(QueryState state, ResourceGroupRuntimeInfo resourceGroupRuntimeInfo)
+    {
+        int queryCount = 0;
+        switch (state) {
+            case RUNNING:
+                queryCount = resourceGroupRuntimeInfo.getDescendantRunningQueries();
+                break;
+            case QUEUED:
+                queryCount = resourceGroupRuntimeInfo.getDescendantQueuedQueries();
+                break;
+            default:
+                fail(format("Unexpected query state %s", state));
+        }
+        return queryCount;
+    }
+
+    private void waitUntilCoordinatorsDiscoveredHealthyInRM(long timeoutInMillis)
+            throws TimeoutException, InterruptedException
+    {
+        long deadline = System.currentTimeMillis() + timeoutInMillis;
+        while (System.currentTimeMillis() < deadline) {
+            AllNodes allNodes = this.resourceManager.refreshNodes();
+            if (allNodes.getActiveCoordinators().size() == COORDINATOR_COUNT) {
+                return;
+            }
+            sleep(100);
+        }
+        throw new TimeoutException(format("one of the nodes is still missing after: %s ms", timeoutInMillis));
     }
 
-    private ClusterStatsResource.ClusterStats getClusterStats(boolean followRedirects)
+    private ClusterStatsResource.ClusterStats getClusterStats(boolean followRedirects, TestingPrestoServer coordinator)
```

Comment:
TestingPrestoServer is not used here, maybe remove?

Reply:
Could you please clarify more on what you meant by TestingPrestoServer is not used here.
getClusterStats returns the cluster stats for the coordinator of type TestingPrestoServer passed as param to this method.

Reply:
OK it seems that it was static field, the rest of the diff was not showing :)